### ISSUE TITLE
Invalid Service Graph Filters?

### ISSUE THREAD

**Issue by bdschaap** [grafana/grafana#80822]
### What happened?

The Service Graph tab for a Tempo data source has filters of `server`, `client`, `__metrics_gen_instance`, `connection_type`, `service`, `span_kind`, `span_name`, and `status_code`.  However, using some of the filters such as `service` cause the Node Graph to be blank.  I suspect the cause may be that the metrics the Node graph uses don't contain a `service` label but am unable to confirm.  The docs state that the traces_service_graph_* metrics contain the `client`, `server`, and `connection_type` labels if those are the metrics that it uses.

### What did you expect to happen?

I expected the Node Graph to work when the available filters are used.

### Did this work before?

Uncertain.

### How do we reproduce it?

1. Configure Tempo for the metrics generator
2. Add a Tempo data source in Grafana and set the prometheus data source where the metrics generator metrics are available.
3. Explore the Tempo data source
4. Go to the Service Graph tab
5. Filter by service
6. The Node Graph will be blank

### Environment (with versions)?

Grafana: 10.2.0
OS: Grafana Linux contain image running on K8s
Browser: Chrome on Windows

### Grafana platform?

Kubernetes

### Datasource(s)?

Tempo

**Comment by 09jvilla** [comment]
I've observed the same problem.

**Comment by 09jvilla** [comment]
<img width="1418" alt="Screenshot 2024-01-23 at 7 07 50 AM" src="https://github.com/grafana/grafana/assets/9610816/c8a90ed5-3b93-4498-bd39-3eef0b9f1000">
<img width="1431" alt="Screenshot 2024-01-23 at 7 08 00 AM" src="https://github.com/grafana/grafana/assets/9610816/0bec856d-2c21-48d9-9fcd-9bf58999d37f">

**Comment by 09jvilla** [comment]
In my example, you'll see that there is a node that is of service `adservice`. But then when I apply a filter for `service=adservice`, I get `no data` for the node graph. It appears that the filter is only working for the table view able the node graph, but not the node graph?

![no-data-node-graph](https://github.com/grafana/grafana/assets/9610816/777bb162-4176-41c1-a0ae-eecc2cd4491a)
![valid-node](https://github.com/grafana/grafana/assets/9610816/4e8becf6-721b-469d-8a20-04bac455d6c3)

**Comment by aocenas** [comment]
I think we should just use tags from tempo there (remove prometheus specific tag names) and then when doing the prom query to get the service graph, replace the tempo tag names with prom specific names so this is ore consistent.

**Comment by 09jvilla** [comment]
I think its a good idea @aocenas as long as we have confidence that the tempo<>prometheus key name mapping will be consistent. 

I followed up with another person in Grafana and she explained to me that this isn't a bug. 

The 'filter' gives you the list of all label names across all Tempo metrics-generator metrics as options to filter on. This means it is the union of the set of labels on the span metrics and the set of labels on the service graph metrics. In my example, I was filtering by a label that was on the span metrics, but not on the service graph metrics, hence why I saw the table populated but didn't see a graph drawn. 

For me then, this isn't a bug even if its confusing.

**Comment by bdschaap** [comment]
Yes I found it odd.  If there was a way to view the query that the node graph runs then it would have been clear.

**Comment by Starefossen** [comment]
RegEx filtering seams to be broken as well.

**Comment by 09jvilla** [comment]
Can you share an example with some screenshots @Starefossen ?

**Comment by Starefossen** [comment]
@09jvilla this seams to have been fixed in latest version of the node graph panel.

Any chance we could get a filter by namespace?

<img width="1009" alt="Screenshot 2024-03-18 at 09 53 03" src="https://github.com/grafana/grafana/assets/968267/62c8541d-61b7-4466-abde-136e3790b8ba">

**Comment by 09jvilla** [comment]
> Any chance we could get a filter by namespace?

Filtering by namespace should work already @Starefossen if you have your metrics-generator configured to output service graph metrics with a `namespace` label.

**Comment by gespi1** [comment]
If you want to filter by the name of a service for spanmetrics and service graphs you can  add `service.name` as additional dimensions on the tempo config 

```
metrics_generator:
      processor:
        service_graphs:
          dimensions:
            - service.name
        span_metrics:
          dimensions:
            - service.name
```
